frontend/src/layout/navigation/navigationLogic.ts

```diff
@@ -222,7 +222,8 @@ export const navigationLogic = kea<navigationLogicType>({
                     return ['demo_project', false]
                 } else if (
                     !user?.is_email_verified &&
-                    featureFlags[FEATURE_FLAGS.REQUIRE_EMAIL_VERIFICATION] === true
+                    !user?.has_social_auth &&
```

Comment:
`user` here comes from `UserSerializer`, not `OrganizationMemberSerializer`, so you might want to put `has_social_auth` on `UserSerializer` too.

But `has_social_auth` seems incomplete for this purpose. It's common for password login to be used alongside social auth, in which case verification should still be needed. It should only be foregone if SSO is enforced at the organization level, as far as I understand.

Reply:
> user here comes from UserSerializer, not OrganizationMemberSerializer, so you might want to put has_social_auth on UserSerializer too.

I believe I did that here? https://github.com/PostHog/posthog/pull/14428/files#diff-78a305cdc1f08147b8c8fc85b69b9d294a0126e54bf084111f89ab7a2a5f203aR61

> But has_social_auth seems incomplete for this purpose. It's common for password login to be used alongside social auth, in which case email verification should still be needed. It should only be foregone if SSO is enforced at the organization level, as far as I understand – that's the only case where we can 100% rely on the external auth provider to ensure the email isn't spoofed.

Hm... We now wipe passwords if someone logs in with social auth, so moving forward that shouldn't happen. But for existing users that could be the case (until they log in with social auth again).

Reply:
Ah, yup, missed that, so it was just a typing issue.

As for `has_social_auth`, I've logged in with social auth multiple times recently and, testing this out with social auth enforcement disabled on Cloud US, I was still able to log in with email address + password – they weren't wiped. So it seems we should check both "has social auth" and "has no password".

Reply:
Did you log in with social login since Monday? I've tested this locally and if your email address is unverified, and you log in with social auth, it should have wiped your account password. If you somehow had a verified email address, then logged in with social auth, it won't have wiped it.

Reply:
I did log in with social login yesterday. But can we just remove the dependency on the password wiping logic from email verification security? It should be a single `is_password_usable()` check if I'm not missing anything

Reply:
> I did log in with social login yesterday.

I looked at your deets in metabase and it looks like you're set with `is_email_verified = true`, so that's why it's not wiping your PW when you log in with SSO. Maybe at some point you tested the verification feature and it set your normal email/pw as verified? I _do_ want to make sure there aren't any security holes here, so if you have any ideas about how this could have happened I'm all ears. In my testing locally and in prod it's not possible to have an unverified email, log in with SSO, and then retain the original PW.

> if SSO is enforced at the organization level, as far as I understand – that's the only case where we can 100% rely on the external auth provider to ensure the email isn't spoofed.

I don't know how SSO being enforced at the org level matters here - can you explain more about that?

--

I promise I'm not trying to be difficult here 😅 it's just that doing this as you suggest isn't that simple - or I'm not quire understanding what you are suggesting (v possible). 

Once someone logs in with social auth, their email address is verified. That's because you _have_ to have access to the email address in order to use social auth (and if they've hijacked the email address that's not our problem / that's what 2FA is for). 

Regardless, once the email address is verified, we have no way of knowing if the _unverified_ password they had set originally was set by _them_ or by someone else trying to gain access to the account (with the register-and-pray-they-use-sso-later method). 

So that's why wiping the passwords is critical. If the email was never verified, and they log in with SSO, we wipe the password so the potential bad actor can't log in with the original password.

If the original email/pw _was_ verified, then the person can keep their PW on the account and use either that or SSO to log in. 

If you want to jump on a call to chat through this then I'm happy to do that tomorrow. Appreciate your help with this 🤗

Reply:
Ah, I made a logical jump earlier, assuming it's possible to link an external provider to an account if their email is different. But that's not the case for us, so it _is_ safe to outsource verification to the provider via login. (I suppose the one case where that doesn't hold is if the user has changed their address on the external account _after_ linking it with PostHog, but it seems extremely unlikely for that to ever be a problem). Sorry for the fuss!